### PR TITLE
Rewrite code entirely

### LINKED ISSUE #12: Prompt becomes "NIL>" when *package* has no nicknames

### LINKED ISSUE #17: Enter with no-input enters the debugger and then fails
Hello,

If we press enter in a void prompt we are brought to the debugger (`SB-IMPL::STRING-INPUT-STREAM`) and with one more we get a `sb-int::STRING-INPUT-STREAM` error

```
[~/bacasable/cl-repl]$ ./cl-repl                                                                       *[master] 
CL-USER> 
Error: end of file on #<SB-IMPL::STRING-INPUT-STREAM {1005626BB3}>
[0]: Try evaluating again.
[1]: Return to top level.
[2]: Edit code.

DEBUG> 

debugger invoked on a END-OF-FILE in thread
#<THREAD "main thread" RUNNING {100560E893}>:
  end of file on #<SB-IMPL::STRING-INPUT-STREAM {1005637633}>

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

(no restarts: If you didn't do this on purpose, please report it as a bug.)

(SB-IMPL::STRING-INCH #<SB-IMPL::STRING-INPUT-STREAM {1005637633}> T 0)
0] 
```

I looked quickly and  couldn't find a place to prevent those.

### PR DESCRIPTION
**Pull request by koji-kojiro**
The source code was entirely rewritten to improve stability and maintainability.
This closes #12, closes #17, closes #23 .

### CODE REVIEW

File: src/main.lisp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+         :description "Print this help and exit."
+         :short #\h
+         :long "help")
+  (:name :version
+         :description "Show the version info and exit."
+         :short #\v
+         :long "version"))
+
+(defun main (argv &key (show-logo t))
+  (multiple-value-bind (options free-args)
```
Comment (vindarel) [inline src/main.lisp]: nice ! I like this way of parsing arguments. If by chance my code helped, I'm glad it did :)

Comment (koji-kojiro) [reply src/main.lisp]: Right, thanks a lot!
However, I could not build an executable with asdf ...
To be accurate, the build succeeded, but the executable did not work, caused SB-SYS: MEMORY-FAULT-ERROR.
I'm not sure where this error comes.
Do you have any ideas?

Comment (vindarel) [reply src/main.lisp]: No, I didn't encounter this error. I had once a "heap exhausted error" and it was a recursion error in a (new-ish) third-party library.

If you put your attempt in a new branch I'm ok to try, and at least say if I reproduce the pb.

Comment (koji-kojiro) [reply src/main.lisp]: OK, please.
I'll create the branch and put comment on #19 later.

**Pull request merged**